test/basic.test.ts

```diff
@@ -2239,41 +2254,63 @@ describe('component islands', () => {
 
     if (isDev()) {
       const fixtureDir = normalize(fileURLToPath(new URL('./fixtures/basic', import.meta.url)))
-      for (const link of result.head.link) {
-        link.href = link.href!.replace(fixtureDir, '/<rootDir>').replaceAll('//', '/')
-        link.key = link.key!.replace(/-[a-z0-9]+$/i, '')
+      for (const head of result.head) {
+        for (const key in head) {
+          if (key === 'link') {
+            head[key] = head[key]?.map((h) => {
+              if (h.href) {
+                h.href = resolveUnrefHeadInput(h.href).replace(fixtureDir, '/<rootDir>').replaceAll('//', '/')
+              }
+              return h
+            })
+          }
+        }
       }
-      result.head.link.sort((a, b) => b.href!.localeCompare(a.href!))
+
+      // todo : why do we need to sort ?
+      // result.head.filter(h => h.tag === 'link').sort((a, b) => b.props.href!.localeCompare(a.props.href!))
```

Comment:
@danielroe why do we needed to sort the links here ?

Reply:
only for test stability - I think I worried that in dev mode vs production they might be in a different order (or maybe it was a race condition?)